chapters/interactions-nonlinear.qmd

```diff
@@ -0,0 +1,893 @@
+
+The observed and predicted visualizations for each model in @tbl-interaction-anova are shown in @fig-lin-reg-interactions. Adding the additional interaction yielded a slight numerical improvement. However, the visualization in the middle panel shows fewer very large residuals. The figure also shows the model results that include the full set of  `r choose(27, 2)` two-way interactions; this panel shows no significant reduction in large residuals, further casting doubt on using the entire set. 
+
+```{r}
+#| label: fig-lin-reg-interactions
+#| echo: false
+#| warning: false
+#| message: false
+#| out-width: 90%
+#| fig-width: 9
+#| fig-height: 3.5
+#| fig-cap: Validation set predicted vs. observed delivery times for linear regression models with and without additional interactions.
+
+lvls <- c("Previous Results", "One Addtional Interaction", "All Item Interactions")
+
+lin_reg_res %>% 
+  collect_predictions() %>% 
+  mutate(Model = "Previous Results") %>% 
+  bind_rows(
+    lin_reg_item_int_res %>% 
+      collect_predictions() %>% 
+      mutate(Model = "One Addtional Interaction")
+  ) %>% 
+  bind_rows(
+    lin_reg_all_int_res %>% 
+      collect_predictions() %>% 
+      mutate(Model = "All Item Interactions")
+  ) %>%
+  mutate(
+    Model = factor(Model, levels = lvls)
+  ) %>%
+  ggplot(aes(time_to_delivery, .pred)) +
+  geom_abline(col = "green", lty = 2) +
+  geom_point(alpha = 0.4, cex = 1) +
+  geom_smooth(se = FALSE, col = "red") +
+  coord_obs_pred() +
+  facet_wrap( ~ Model, ncol = 3) +
+  labs(x = "Observed Delivery Time", y = "Predicted Delivery Time")
+```
+
+So far, we have used visualizations to unearth potential interactions. This can be an effective strategy when the number of potential interactions is large, but the visual nature of this process is subjective. There are some specialized quantitative tools for identifying interactions. For example, @lim2015learning used regularized generalized linear models (sections TODO) to estimate all possible two-way interactions and use a penalization method to determine which should be retained. @miller1984selection and @fes ([Section 7.4.3](https://bookdown.org/max/FES/approaches-when-complete-enumeration-is-practically-impossible.html#the-feasible-solution-algorithm)) describe the feasible solution algorithm, an iterative search method for interaction discovery. Another, which we will now describe in more detail, is Friedman's H-statistic [@friedman2008predictive]. 
```

Comment:
@kjell-stattenacity is `miller1984selection` the best reference for FSA and interactions?

Reply:
I need to check on this

Reply:
Added a new issue for this to be able to merge this PR